### PR TITLE
DZ LESSON3

### PR DESCRIPTION
**Pull request by NEULOVIM**

### CODE REVIEW

**Comment by NEULOVIM** [comment]
Sorry что не по порядку и не сразу одним. В следующий раз буду знать что лучше одним и сразу всё.

File: src/lesson3/Task1_1.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Введите число любого месяца");
+        String a = sc.nextLine();
+
+        switch (a) {
+            case "1":
+                System.out.println("Январь");
```
Comment (DHrytsuk) [inline src/lesson3/Task1_1.java]: We need to output the season, not the month

File: src/lesson3/Task1_4.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public class Task1_4 {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        int t = sc.nextInt();
+
+        if (t > -5){
+            System.out.println("Тепло");
+        } else if (t <= -5 || t > -20) {
```
Comment (DHrytsuk) [inline src/lesson3/Task1_4.java]: It looks not correct
E.g. enter -30

File: src/lesson3/Task1_5.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            case "3":
+                System.out.println("Жёлтый");
+                break;
+            case "4":
+                System.out.println("Зелёный");
+            case "5":
+                System.out.println("Голоубой");
+                break;
+            case "6":
+                System.out.println("Синий");
```
Comment (DHrytsuk) [inline src/lesson3/Task1_5.java:30]: `break;`?

File: src/lesson3/_Cikl/Task1.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package lesson3._Cikl;
+
+public class Task1 {
+    public static void main(String[] args) {
+
+        for (int i = 0; i <= 99; i++) {
```
Comment (DHrytsuk) [inline src/lesson3/_Cikl/Task1.java]: It make sense to start loop from 1, according to the task

File: src/lesson3/_Cikl/Task2.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+
+        int a = sc.nextInt();
+        int sum = 1;
+
+        for (int i = 1; i < a; i++){
+            sum = sum + i;
```
Comment (DHrytsuk) [inline src/lesson3/_Cikl/Task2.java:15]: Looks loke the result will be incorrect

File: src/lesson3/_Cikl/Task3.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package lesson3._Cikl;
+
+public class Task3 {
+    public static void main(String[] args) {
+
+        int g = 7;
+        while (g <= 99) {
+            System.out.println(g);
+            g ++;
```
Comment (DHrytsuk) [inline src/lesson3/_Cikl/Task3.java]: It's very bad approach!!!
Show me how did you try with `g + 7`

File: src/lesson3/_Cikl/Task4.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package lesson3._Cikl;
+
+public class Task4 {
+    public static void main(String[] args) {
+        for (int i = 1; i <= 20; i++){
```
Comment (DHrytsuk) [inline src/lesson3/_Cikl/Task4.java]: Please, read the task carefully.
We need to start from 10

File: src/lesson3/Task1_1.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Введите число любого месяца");
+        String a = sc.nextLine();
+
+        switch (a) {
+            case "1":
+                System.out.println("Зима");
```
Comment (DHrytsuk) [inline src/lesson3/Task1_1.java]: Can you optimize it to avoid code duplicates?

File: src/lesson3/Task1_2.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package lesson3;
+
+import java.util.Scanner;
+
+public class Task1_2 {
+    public static void main(String[] args) {
+
+//Scanner sc = new Scanner(System.in);    не хочит работать через сканер
```
Comment (DHrytsuk) [inline src/lesson3/Task1_2.java:8]: Is there any exception?

File: src/lesson3/Task1_5.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+                System.out.println("Красный");
+                break;
+            case "2":
+                System.out.println("Оранжевый");
+                break;
+            case "3":
+                System.out.println("Жёлтый");
+                break;
+            case "4":
+                System.out.println("Зелёный");
```
Comment (DHrytsuk) [inline src/lesson3/Task1_5.java:24]: `break;`?

File: src/lesson3/_Cikl/Task3.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package lesson3._Cikl;
+
+public class Task3 {
+    public static void main(String[] args) {
+
+        int g = 7;
+        while (g <= 99) {
+            System.out.println(g);
+            g = g + 7;
```
Comment (DHrytsuk) [inline src/lesson3/_Cikl/Task3.java:9]: Is it working now?

File: src/lesson3/Additionaly/ADD3.java [hunk 12]
```diff
@@ -0,0 +1,71 @@
+package lesson3.Additionaly;
+
+public class ADD3 {
+    public static void main(String[] args) {
+
+
+        for (int a1 = 1; a1 <= 10; a1++) {
```
Comment (DHrytsuk) [inline src/lesson3/Additionaly/ADD3.java]: Very very strange!!!
1. Why do you use separate loop for each number?
2. Why do you use `break`, if ypu can change the loop condition a bit?
3. Formating

File: src/lesson3/_Cikl/Task3.java (hunk 11 above)
Comment (NEULOVIM) [reply src/lesson3/_Cikl/Task3.java:9]: yeas, it's working.
Task3

File: src/lesson3/Task1_5.java (hunk 10 above)
Comment (NEULOVIM) [reply src/lesson3/Task1_5.java:24]: Woth break? I didn't understand

File: src/lesson3/Additionaly/ADD3.java
Comment (NEULOVIM) [inline src/lesson3/Additionaly/ADD3.java:1]: uploaded by mistake (ADD3)

**Review by DHrytsuk — APPROVED** [review]

File: src/lesson3/Additionaly/ADD2.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+        float sum = sc.nextFloat();
+
+        float b = (sum * 7 / 100) + sum;    // число c %, за 1 месяц.
+
+        System.out.println("Введите количество месяцев");
+        int m = sc.nextInt();
+
+
+        for (int i = 1; i <= m; i++) {
+            float finish =  b * i;
```
Comment (DHrytsuk) [inline src/lesson3/Additionaly/ADD2.java]: Is it working correct?
it seems, if the initial amount will be 100 and month will be 3, then as a result you'll have more than 300, which is incorrect

File: src/lesson3/Additionaly/ADD2.java [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        float sum = sc.nextFloat();
+
+        System.out.println("Введите количество месяцев");
+        float math = sc.nextInt();
+
+
+        float sumFinish = (sum * 7 / 100) + sum;    // число + %, за 1 месяц.
+
+        int i = 0;
+        for (; i < math; i++)
```
Comment (DHrytsuk) [inline src/lesson3/Additionaly/ADD2.java]: What is it?
Still not correct

File: src/lesson3/Additionaly/ADD2.java [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+        {
+        }
+
+        float finish = sumFinish * i + sum;
+        System.out.println(finish);
+
+//        Почему сначала цикл не отработает а потом не прочтёт то что за циклом?
```
Comment (DHrytsuk) [inline src/lesson3/Additionaly/ADD2.java]: You have an empty loop and it's strange!

**Pull request merged**